clang/test/ClangScanDeps/response-file-clang-cl.c

```diff
@@ -0,0 +1,36 @@
+// Check that the scanner can adjust arguments by reading .rsp files in advance.
+
+// RUN: rm -rf %t
+// RUN: split-file %s %t
+// RUN: sed -e "s|DIR|%/t|g" %t/cdb.json.template > %t/cdb.json
```

Comment:
You can now do this: `clang-scan-deps -o %t/deps.json -- %clang_cl <args>`. I guess `<args>` for `clang-cl` specifically cannot contain `%t` since that will begin with `/` on Unix system and cause command line parsing errors. But using relative paths like you currently do and changing directory before running the tools should work a will probably be much cleaner.

Reply:
As suggested. We're taking a different codepath when using the immediate mode (using `InplaceCompilationDatabase`) and that expands the response files in `OptTable::parseArgs()` so I wasn't really covering all the changes in the patch. I've added tests for both cases. The immediate mode allowed me to uncover a bug in `ParseArgs()`: `Saver` and `Alloc` were local to the function but we were using their contents to populate `CommandLine`. I fixed that too.

Reply:
Ok, that makes sense, thanks!